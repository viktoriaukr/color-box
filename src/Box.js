import React from "react";

const Box = ({ id, width = 6, height = 6, backgroundColor, remove }) => {
  const removeBox = () => remove(id);
  return (
    <div className="Box-container">
      <div
        className="Box"
        style={{
          width: `${width}em`,
          height: `${height}em`,
          backgroundColor,
        }}
      ></div>
      <button className="Box-btn" onClick={removeBox}>
        X
      </button>
    </div>
  );
};

export default Box;
