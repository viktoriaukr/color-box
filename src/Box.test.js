import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

test("render", () => {
  render(<Box />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
