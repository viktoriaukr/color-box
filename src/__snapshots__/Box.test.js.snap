// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box-container"
  >
    <div
      class="Box"
      style="width: 6em; height: 6em;"
    />
    <button
      class="Box-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
